mobile_verifier/src/unique_connections/ingestor.rs

```diff
@@ -0,0 +1,181 @@
+                .lookback(LookbackBehavior::StartAfter(settings.start_after))
+                .prefix(FileType::UniqueConnectionsReport.to_string())
+                .create()
+                .await?;
+
+        let radio_threshold_ingestor = Self::new(
+            pool.clone(),
+            unique_connections_ingest,
+            verified_unique_connections,
+            authorization_verifier,
+        );
+
+        Ok(TaskManager::builder()
+            .add_task(verified_unique_conections_server)
+            .add_task(radio_threshold_ingestor)
+            .add_task(unique_connections_server)
+            .build())
+    }
+
+    pub fn new(
+        pool: PgPool,
+        unique_connections_receiver: Receiver<FileInfoStream<UniqueConnectionsIngestReport>>,
+        verified_unique_connections_sink: FileSinkClient<VerifiedUniqueConnectionsIngestReportV1>,
+        authorization_verifier: AV,
+    ) -> Self {
+        Self {
+            pool,
+            unique_connections_receiver,
+            verified_unique_connections_sink,
+            authorization_verifier,
+        }
+    }
+
+    async fn run(mut self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("starting unique connections ingestor");
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => break,
+                Some(file) = self.unique_connections_receiver.recv() => {
+                    self.process_unique_connections_file(file).await?;
+                }
+            }
+        }
+        tracing::info!("stopping unique connections ingestor");
+        Ok(())
+    }
+
+    async fn process_unique_connections_file(
+        &self,
+        file_info_stream: FileInfoStream<UniqueConnectionsIngestReport>,
+    ) -> anyhow::Result<()> {
+        let mut txn = self.pool.begin().await?;
+
+        let mut stream = file_info_stream.into_stream(&mut txn).await?;
+        while let Some(unique_connections_report) = stream.next().await {
+            let verified_report_status = self
+                .verify_unique_connection_report(&unique_connections_report.report)
+                .await;
+
+            if matches!(
+                verified_report_status,
+                VerifiedUniqueConnectionsIngestReportStatus::Valid
+            ) {
+                db::save(&mut txn, &unique_connections_report).await?;
+            }
+
+            let verified_report_proto = VerifiedUniqueConnectionsIngestReport {
+                timestamp: Utc::now(),
+                report: unique_connections_report,
+                status: verified_report_status,
+            };
+
+            self.verified_unique_connections_sink
+                .write(
+                    verified_report_proto.into(),
+                    &[("report_status", verified_report_status.as_str_name())],
+                )
+                .await?;
+        }
+
+        txn.commit().await?;
```

Comment:
without an on conflict on the DB save, there is a risk of inserting the same data multiple times should the same file end up being reprocessed